packages/truffle-blockchain-utils/index.js

```diff
@@ -1,28 +1,22 @@
-const Blockchain = {
-  getBlockByNumber(blockNumber, provider, callback) {
-    const params = [blockNumber, true];
-    provider.send(
-      {
-        jsonrpc: "2.0",
-        method: "eth_getBlockByNumber",
-        params,
-        id: Date.now()
-      },
-      callback
-    );
-  },
+const defaultJSONQuery = {
+  jsonrpc: "2.0",
+  method: "",
+  params: [],
+  id: Date.now()
+};
 
-  getBlockByHash(blockHash, provider, callback) {
-    const params = [blockHash, true];
-    provider.send(
-      {
-        jsonrpc: "2.0",
-        method: "eth_getBlockByHash",
-        params,
-        id: Date.now()
-      },
-      callback
-    );
+const Blockchain = {
+  getBlockBy(NumberOrHash, value, provider, callback) {
+    defaultJSONQuery.params = [value, true];
+    if (NumberOrHash === "Number")
+      defaultJSONQuery.method = "eth_getBlockByNumber";
+    else if (NumberOrHash === "Hash")
+      defaultJSONQuery.method = "eth_getBlockByHash";
+    else
+      throw new Error(
+        `.getBlockBy() only currently supports querying a "Number" or "Hash"!`
+      );
```

Comment:
I'd push back on having string-denoted things that alter a function's behaviour. A bit of a code smell IMO. I actually think the duplication here is not that big a deal and it's actually more complex and less readable than before.